React/Views/RCTHandledKey.h

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#if TARGET_OS_OSX // [macOS
+#import <React/RCTConvert.h>
+
+@interface RCTHandledKey : NSObject
+
++ (BOOL)event:(NSEvent *)event matchesFilter:(NSArray<RCTHandledKey *> *)filter;
++ (BOOL)key:(NSString *)key matchesFilter:(NSArray<RCTHandledKey *> *)filter;
+
+- (instancetype)initWithKey:(NSString *)key;
+- (BOOL)matchesEvent:(NSEvent *)event;
+
+@property (nonatomic, copy) NSString *key;
+@property (nonatomic, assign) NSNumber *capsLock; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *shift; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *ctrl; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *alt; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *meta; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *numericPad; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *help; // boolean; nil == don't care
+@property (nonatomic, assign) NSNumber *function; // boolean; nil == don't care
```

Comment:
NSNumber to essentially simulate an optional Bool? But if we omit the "don't care modifiers" bit, then these can just be a BOOL?

Reply:
that's correct -- if we decide to remove the "don't care modifiers", then we can use plain BOOLs. However, that will break the legacy behavior where validKeys[Down|Up] didn't care about modifiers either. I say we keep this so we can have some way of preserving old behavior

Reply:
Since we're keeping the legacy behavior of ignoring modifiers if the passed in key is a string and not a HandledKey-type object, we're going to need to leave this as is.